javascripts/controllers/NewCtrl.js

```diff
@@ -1,5 +1,37 @@
 "use strict";
 
-app.controller("NewCtrl", function($scope){
-    $scope.controller = "NewCtrl";
-});
\ No newline at end of file
+app.controller("NewCtrl", function($location, $rootScope, $scope, ContactsService){
+
+    let userId = $rootScope.uid;
+    const createContactObject = (contact) => {
+        return{
+            "first_name": contact.first_name,
+            "last_name": contact.last_name,
+            "title": contact.title,
+            "company": contact.company,
+            "category": contact.category,
+            "cell_phone": contact.cell_phone,
+            "email": contact.cell_phone,
+            "location": contact.location,
+            "github_link": contact.github_link,
+            "linkedin_link": contact.linkedin_link,
+            "image": contact.image,
+            "favorite": false,
+            "uid": userId
+        };
+    };
+
+    $scope.addNewContact = (contact) => {
+        let newContact = createContactObject(contact);
+        console.log("newContact", newContact);
+        ContactsService.postNewContact(newContact, contact.id).then((results) => {
```

Comment:
more stuff should be coming from your form or you will not have every attribute you need in createContactObject